xdem/volume.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+
+def hypsometric_binning(ddem: np.ndarray, dem: np.ndarray):
+
+    assert ddem.shape == dem.shape
+    nan_mask = np.isnan(ddem) | np.isnan(dem)
+    ddem = ddem[~nan_mask]
+    dem = dem[~nan_mask]
+
+    old_dem = dem - ddem
+    elevation_range = np.percentile(old_dem, 99) - np.percentile(old_dem, 1)
+
+    bin_size = 50 if elevation_range > 500 else elevation_range / 10
```

Comment:
maybe we can move this into an argument, with a default setting

Reply:
Added this in 786b385. It can either be in georeferenced units or as a percentage (e.g. 10). Please let me know what you think of the argument names and the structure!